docs/src/main/asciidoc/redis.adoc

```diff
@@ -0,0 +1,547 @@
+
+== Creating the Maven Project
+
+First, we need a new project. Create a new project with the following command:
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=redis-quickstart \
+    -Dextensions="redis-client, resteasy-jsonb, resteasy-mutiny"
+cd redis-quickstart
+----
+
+This command generates a Maven project, importing the Redis extension.
+
+
+If you already have your Quarkus project configured, you can add the `redis-client` extension
+to your project by running the following command in your project base directory:
+
+[source,bash]
+----
+./mvnw quarkus:add-extension -Dextensions="redis-client"
+----
+
+This will add the following to your `pom.xml`:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-redis-client</artifactId>
+</dependency>
+----
+
+
+== Starting the Redis server
+
+Then, we need to start a Redis instance (if you do not have one already) using the following command:
+
+[source, bash]
+----
+docker run --ulimit memlock=-1:-1 -it --rm=true --memory-swappiness=0 --name redis_quarkus_test -p 6379:6379 redis:5.0.6
+----
+
+== Configuring Redis properties
+
+Once we have the Redis server running, we need to configure the Redis connection properties.
+This is done in the `application.properties` configuration file. Edit it to the following content:
+
+[source]
+----
+quarkus.redis.hosts=localhost:6379 <1>
+----
+
+1. Configure Redis hosts to connect to. Here we connect to the Redis server we started in the previous section
+
+== Creating the Increment POJO
+
+We are going to model our increments using the `Increment` POJO.
+Create the `src/main/java/org/acme/redis/Increment.java` file, with the following content:
+
+[source, java]
+----
+package org.acme.redis;
+
+public class Increment {
+    public String key; <1>
+    public int value; <2>
+
+    public Increment(String key, int value) {
+        this.key = key;
+        this.value = value;
+    }
+
+    public Increment() {
+    }
+}
+----
+
+1. The key that will be used as the Redis key
+2. The value held by the Redis key
+
+
+== Creating the Increment Service
+
+We are going to create an `IncrementService` class which will play the role of a Redis client.
+With this class, we'll be able to perform the `SET`, `GET` , `DELET`, `KEYS` and `INCRBY` Redis commands.
+
+Create the `src/main/java/org/acme/redis/IncrementService.java` file, with the following content:
+
+[source, java]
+----
+package org.acme.redis;
+
+import io.quarkus.redis.client.SyncRedisAPI;
+import io.smallrye.mutiny.Uni;
+
+import io.vertx.mutiny.redis.client.Response;
+import io.vertx.mutiny.redis.client.RedisAPI;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+class IncrementService {
+
+    @Inject
+    SyncRedisAPI syncRedisAPI; <1>
```

Comment:
It is sync and blocking since it is a synchronous client. Okay with the renaming to `RedisClient` and I'll also a `ReactiveRedisClient` (which was not there as I expected users to be using the mutiny based one).